test/fixtures/types/test-esm-types.ts

```diff
@@ -0,0 +1,30 @@
+// Test ESM TypeScript imports with Top-Level Await
+import { Client, Server, Message, Bundle, encode, decode } from '../../../lib/index.mjs';
+
+// Create server first (typical usage pattern)
+const server: Server = new Server(3333, '0.0.0.0');
```

Comment:
can you await server being available, you can see the pattern in the examples folder.